app/src/main/java/com/example/ncov19traking/ui/CovidCustomCard.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.ncov19traking.ui
+
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import com.example.ncov19traking.R
+import com.google.android.material.card.MaterialCardView
+
+const val DEFAULT_TITLE_TEXT = "Title"
+const val DEFAULT_NUMBER_TEXT = "0"
+const val DEFAULT_PERCENTAGE_TEXT = "0%"
+const val CARD_ELEVATION = 10F
+const val CARD_RADIUS = 16F
+
+class CovidCustomCard @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : MaterialCardView(context, attrs, defStyleAttr) {
+
+    private var titleText = DEFAULT_TITLE_TEXT
+    private var numberText = DEFAULT_NUMBER_TEXT
+    private var percentageText = DEFAULT_PERCENTAGE_TEXT
+    private var numberTextColor = resources.getColor(R.color.recoveredColor)
+    private var percentageTextColor = resources.getColor(R.color.textColor)
+    private lateinit var titleTextView : TextView
+    private lateinit var numbersTextView : TextView
+    private lateinit var percentageTextView : TextView
```

Comment:
if you put your `findByIds` at the end of the init block you can stop using late init var here that are not bad but are expensive, and in this custom view you are not gaining anything by using them.

Reply:
I used lateinit to have the view available for the setters. May I ask you for another way to do this? so I can put it in practice